samples/wavlib/src/main/cpp/io/wav/WavFmtChunkHeader.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <android/log.h>
+
+#include "io/stream/InputStream.h"
+
+#include "WavFmtChunkHeader.h"
+
+static const char *TAG = "WavFmtChunkHeader";
+
+namespace wavlib {
+
+const RiffID WavFmtChunkHeader::RIFFID_FMT = makeRiffID('f', 'm', 't', ' ');
+
+WavFmtChunkHeader::WavFmtChunkHeader() : WavChunkHeader(RIFFID_FMT) {
+    mFormatId = ENCODING_PCM;
+    mNumChannels = 0;
+    mSampleRate = 0;
+    mAveBytesPerSecond = 0;
+    mBlockAlign = 0;
+    mSampleSize = 0;
+    mExtraBytes = 0;
+}
+
+WavFmtChunkHeader::WavFmtChunkHeader(RiffID tag) : WavChunkHeader(tag) {
+    mFormatId = ENCODING_PCM;
+    mNumChannels = 0;
+    mSampleRate = 0;
+    mAveBytesPerSecond = 0;
+    mBlockAlign = 0;
+    mSampleSize = 0;
+    mExtraBytes = 0;
+}
+
+void WavFmtChunkHeader::normalize() {
+    if (mFormatId == ENCODING_PCM || mFormatId == ENCODING_IEEE_FLOAT) {
+        mBlockAlign = (short) (mNumChannels * (mSampleSize / 8));
+        mAveBytesPerSecond = mSampleRate * mBlockAlign;
+        mExtraBytes = 0;
+    } else {
+        //hmmm....
+    }
+}
+
+void WavFmtChunkHeader::read(InputStream *stream) {
+    WavChunkHeader::read(stream);
+    stream->read(&mFormatId, sizeof(mFormatId));

```

Comment:
Assumes Little Endian. This maybe a safe assumption for Android. But maybe not forever.
Maybe assert in the constructor if BigEndian CPU.
Or call readLittleEndian(int16_t *shortData). Don't need a size.